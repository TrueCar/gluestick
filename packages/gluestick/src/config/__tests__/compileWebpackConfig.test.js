/* @flow */
jest.mock('../webpack/buildEntries.js', () => () => ({}));
jest.mock('../webpack/buildServerEntries.js', () => jest.fn());
jest.mock('../webpack/prepareEntries.js', () => jest.fn());

const compileWebpackConfig = require('../compileWebpackConfig');
const defaultGSConfig = require('../defaults/glueStickConfig');

const loggerMock = {
  info: jest.fn(),
  debug: jest.fn(),
  success: jest.fn(),
  error: jest.fn(),
};

describe('config/compileWebpackConfig', () => {
  it('should return webpack config', () => {
    // $FlowIgnore
    const webpackConfig = compileWebpackConfig(loggerMock, [
      {
        preOverwrites: {},
        postOverwrites: {
          // $FlowIgnore
          clientWebpackConfig: (config) => Object.assign(config, { testProp: true }),
          // $FlowIgnore
          serverWebpackConfig: (config) => Object.assign(config, { testProp: true }),
        },
      },
      {
        preOverwrites: {
          // $FlowIgnore
          sharedWebpackConfig: (config) => Object.assign(config, { preTestProp: true }),
        },
        postOverwrites: {
          // $FlowIgnore
          clientWebpackConfig: (config) => Object.assign(config, { testPropNew: true }),
        },
      },
    ], {}, defaultGSConfig);
    expect(webpackConfig.universalSettings).not.toBeNull();
    expect(webpackConfig.client).not.toBeNull();
    expect(webpackConfig.server).not.toBeNull();
    expect(webpackConfig.client.testProp).toBeTruthy();
    expect(webpackConfig.server.testProp).toBeTruthy();
    expect(webpackConfig.client.preTestProp).toBeTruthy();
    expect(webpackConfig.server.preTestProp).toBeTruthy();
    expect(webpackConfig.client.testPropNew).toBeTruthy();
  });
});
