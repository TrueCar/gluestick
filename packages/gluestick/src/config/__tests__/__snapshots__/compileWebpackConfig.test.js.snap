// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config/compileWebpackConfig in dev environment returns the correct client config 1`] = `
Object {
  "context": "/project",
  "devtool": "cheap-module-source-map",
  "entry": Object {},
  "module": Object {
    "rules": Array [
      Object {
        "exclude": /\\(node_modules\\\\/\\(\\?!gluestick\\)\\)\\|\\(build\\\\/assets\\)/,
        "test": /\\\\\\.js\\$/,
        "use": Array [
          Object {
            "loader": "babel-loader",
            "options": Object {
              "babelrc": false,
              "plugins": Array [
                "transform-flow-strip-types",
                "react-hot-loader/babel",
              ],
              "presets": Array [
                Array [
                  "es2015",
                  Object {
                    "modules": false,
                  },
                ],
                "react",
                "stage-0",
              ],
            },
          },
        ],
      },
      Object {
        "test": /\\\\\\.\\(s\\?css\\)\\$/,
        "use": Array [
          "style-loader",
          Object {
            "loader": "css-loader",
            "options": Object {
              "minimize": false,
              "sourceMap": false,
            },
          },
          Object {
            "loader": "postcss-loader",
          },
          Object {
            "loader": "sass-loader",
            "options": Object {
              "outputStyle": "expanded",
              "sourceMap": false,
              "sourceMapContents": true,
            },
          },
        ],
      },
      Object {
        "test": /\\\\\\.\\(png\\|jpg\\|gif\\|ico\\|svg\\)\\(\\\\\\?v=\\\\d\\+\\\\\\.\\\\d\\+\\\\\\.\\\\d\\+\\)\\?\\$/,
        "use": Array [
          Object {
            "loader": "file-loader",
            "options": Object {
              "name": "[name]-[hash].[ext]",
            },
          },
          Object {
            "loader": "image-webpack-loader",
            "options": Object {},
          },
        ],
      },
      Object {
        "test": /\\\\\\.\\(woff\\|woff2\\|ttf\\|eot\\)\\(\\\\\\?v=\\\\d\\+\\\\\\.\\\\d\\+\\\\\\.\\\\d\\+\\)\\?\\$/,
        "use": Array [
          Object {
            "loader": "file-loader",
            "options": Object {
              "name": "[name]-[hash].[ext]",
            },
          },
        ],
      },
    ],
  },
  "node": Object {
    "net": "empty",
  },
  "output": Object {
    "chunkFilename": "[name].[hash].js",
    "crossOriginLoading": "anonymous",
    "filename": "[name].[hash].js",
    "path": "/project/build/assets",
    "publicPath": "http://0.0.0.0:8888/assets/",
  },
  "performance": Object {
    "hints": false,
  },
  "plugins": Array [
    NormalModuleReplacementPlugin {
      "newResource": "/project/packages/gluestick/src/config/webpack/mocks/nullMock.js",
      "resourceRegExp": /\\\\\\.server\\(\\\\\\.js\\)\\?\\$/,
    },
    DuplicatePackageCheckerPlugin {
      "options": Object {
        "emitError": false,
        "exclude": null,
        "verbose": false,
      },
    },
    ProgressPlugin {
      "handler": undefined,
      "profile": undefined,
    },
    CommonsChunkPlugin {
      "async": undefined,
      "children": undefined,
      "chunkNames": Array [
        "vendor",
      ],
      "deepChildren": undefined,
      "filenameTemplate": "vendor.bundle.js",
      "ident": "/project/packages/gluestick/node_modules/webpack/lib/optimize/CommonsChunkPlugin.js0",
      "minChunks": undefined,
      "minSize": undefined,
      "selectedChunks": undefined,
    },
    ChunksPlugin {
      "configuration": Object {
        "context": "/project",
        "entry": Object {},
        "module": Object {
          "rules": Array [
            Object {
              "exclude": /\\(node_modules\\\\/\\(\\?!gluestick\\)\\)\\|\\(build\\\\/assets\\)/,
              "test": /\\\\\\.js\\$/,
              "use": Array [
                Object {
                  "loader": "babel-loader",
                  "options": Object {
                    "babelrc": false,
                    "plugins": Array [
                      "transform-flow-strip-types",
                    ],
                    "presets": Array [
                      Array [
                        "es2015",
                        Object {
                          "modules": false,
                        },
                      ],
                      "react",
                      "stage-0",
                    ],
                  },
                },
              ],
            },
            Object {
              "test": /\\\\\\.\\(s\\?css\\)\\$/,
              "use": Array [
                "style-loader",
                Object {
                  "loader": "css-loader",
                  "options": Object {
                    "minimize": false,
                    "sourceMap": false,
                  },
                },
                Object {
                  "loader": "postcss-loader",
                },
                Object {
                  "loader": "sass-loader",
                  "options": Object {
                    "outputStyle": "expanded",
                    "sourceMap": false,
                    "sourceMapContents": true,
                  },
                },
              ],
            },
            Object {
              "test": /\\\\\\.\\(png\\|jpg\\|gif\\|ico\\|svg\\)\\(\\\\\\?v=\\\\d\\+\\\\\\.\\\\d\\+\\\\\\.\\\\d\\+\\)\\?\\$/,
              "use": Array [
                Object {
                  "loader": "file-loader",
                  "options": Object {
                    "name": "[name]-[hash].[ext]",
                  },
                },
                Object {
                  "loader": "image-webpack-loader",
                  "options": Object {},
                },
              ],
            },
            Object {
              "test": /\\\\\\.\\(woff\\|woff2\\|ttf\\|eot\\)\\(\\\\\\?v=\\\\d\\+\\\\\\.\\\\d\\+\\\\\\.\\\\d\\+\\)\\?\\$/,
              "use": Array [
                Object {
                  "loader": "file-loader",
                  "options": Object {
                    "name": "[name]-[hash].[ext]",
                  },
                },
              ],
            },
          ],
        },
        "node": Object {
          "net": "empty",
        },
        "output": Object {
          "chunkFilename": "[name].[hash].js",
          "filename": "[name].[hash].js",
          "path": "/project/build/assets",
          "publicPath": "/assets/",
        },
        "plugins": Array [
          NormalModuleReplacementPlugin {
            "newResource": "/project/packages/gluestick/src/config/webpack/mocks/nullMock.js",
            "resourceRegExp": /\\\\\\.server\\(\\\\\\.js\\)\\?\\$/,
          },
          DuplicatePackageCheckerPlugin {
            "options": Object {
              "emitError": false,
              "exclude": null,
              "verbose": false,
            },
          },
          ProgressPlugin {
            "handler": undefined,
            "profile": undefined,
          },
          CommonsChunkPlugin {
            "async": undefined,
            "children": undefined,
            "chunkNames": Array [
              "vendor",
            ],
            "deepChildren": undefined,
            "filenameTemplate": "vendor.bundle.js",
            "ident": "/project/packages/gluestick/node_modules/webpack/lib/optimize/CommonsChunkPlugin.js0",
            "minChunks": undefined,
            "minSize": undefined,
            "selectedChunks": undefined,
          },
        ],
        "resolve": Object {
          "alias": Object {
            "apps": "/project/src/apps",
            "assets": "/project/assets",
            "compiled": "/project/node_modules",
            "config": "/project/src/config",
            "root": "/project",
            "shared": "/project/src/shared",
            "src": "/project/src",
          },
          "extensions": Array [
            ".js",
            ".css",
            ".json",
          ],
        },
      },
      "options": Object {},
    },
    DefinePlugin {
      "definitions": Object {
        "process.env.NODE_ENV": "\\"development\\"",
      },
    },
    HotModuleReplacementPlugin {
      "fullBuildTimeout": 200,
      "multiStep": undefined,
      "options": Object {},
      "requestTimeout": 10000,
    },
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "resolve": Object {
    "alias": Object {
      "apps": "/project/src/apps",
      "assets": "/project/assets",
      "compiled": "/project/node_modules",
      "config": "/project/src/config",
      "root": "/project",
      "shared": "/project/src/shared",
      "src": "/project/src",
    },
    "extensions": Array [
      ".js",
      ".css",
      ".json",
    ],
  },
}
`;

exports[`config/compileWebpackConfig in dev environment returns the correct server config 1`] = `
Object {
  "context": "/project",
  "devtool": "source-map",
  "entry": Object {
    "renderer": "/project/packages/gluestick/src/renderer/entry.js",
  },
  "externals": Array [
    [Function],
  ],
  "module": Object {
    "noParse": Array [
      /cli\\\\/helpers/,
    ],
    "rules": Array [
      Object {
        "exclude": /\\(node_modules\\\\/\\(\\?!gluestick\\)\\)\\|\\(build\\\\/assets\\)/,
        "test": /\\\\\\.js\\$/,
        "use": Array [
          Object {
            "loader": "babel-loader",
            "options": Object {
              "babelrc": false,
              "plugins": Array [
                "transform-flow-strip-types",
              ],
              "presets": Array [
                "es2015",
                "react",
                "stage-0",
              ],
            },
          },
        ],
      },
      Object {
        "test": /\\\\\\.\\(s\\?css\\)\\$/,
        "use": Array [
          "ignore-loader",
        ],
      },
      Object {
        "test": /\\\\\\.\\(png\\|jpg\\|gif\\|ico\\|svg\\)\\(\\\\\\?v=\\\\d\\+\\\\\\.\\\\d\\+\\\\\\.\\\\d\\+\\)\\?\\$/,
        "use": Array [
          Object {
            "loader": "file-loader",
            "options": Object {
              "emitFile": false,
              "name": "[name]-[hash].[ext]",
            },
          },
          Object {
            "loader": "image-webpack-loader",
            "options": Object {},
          },
        ],
      },
      Object {
        "test": /\\\\\\.\\(woff\\|woff2\\|ttf\\|eot\\)\\(\\\\\\?v=\\\\d\\+\\\\\\.\\\\d\\+\\\\\\.\\\\d\\+\\)\\?\\$/,
        "use": Array [
          Object {
            "loader": "file-loader",
            "options": Object {
              "emitFile": false,
              "name": "[name]-[hash].[ext]",
            },
          },
        ],
      },
    ],
  },
  "node": Object {
    "__dirname": false,
    "__filename": false,
    "net": "empty",
  },
  "output": Object {
    "chunkFilename": "[name].js",
    "filename": "[name].js",
    "libraryTarget": "commonjs2",
    "path": "/project/build/server",
    "pathinfo": true,
    "publicPath": "http://localhost:8888undefined",
  },
  "plugins": Array [
    ProgressPlugin {
      "handler": undefined,
      "profile": undefined,
    },
    DefinePlugin {
      "definitions": Object {
        "process.env.NODE_ENV": "\\"development\\"",
      },
    },
    LimitChunkCountPlugin {
      "options": Object {
        "maxChunks": 1,
      },
    },
    BannerPlugin {
      "banner": "require(\\"source-map-support\\").install();",
      "options": Object {
        "banner": "require(\\"source-map-support\\").install();",
        "entryOnly": false,
        "raw": true,
      },
    },
  ],
  "resolve": Object {
    "alias": Object {
      "application-config": "/project/src/config/application",
      "apps": "/project/src/apps",
      "assets": "/project/assets",
      "compiled": "/project/node_modules",
      "config": "/project/src/config",
      "entry-wrapper": "/project/gluestick/EntryWrapper",
      "gluestick-hooks": "/project/src/gluestick.hooks.js",
      "plugins-config-path": undefined,
      "project-entries": "/project/gluestick/entries",
      "project-entries-config": "/project/src/entries.json",
      "redux-middlewares": "/project/src/config/redux-middleware",
      "root": "/project",
      "shared": "/project/src/shared",
      "src": "/project/src",
    },
    "extensions": Array [
      ".js",
      ".css",
      ".json",
    ],
  },
  "target": "node",
}
`;

exports[`config/compileWebpackConfig in production environment returns the correct client config 1`] = `
Object {
  "bail": true,
  "context": "/project",
  "devtool": "hidden-source-map",
  "entry": Object {},
  "module": Object {
    "rules": Array [
      Object {
        "exclude": /\\(node_modules\\\\/\\(\\?!gluestick\\)\\)\\|\\(build\\\\/assets\\)/,
        "test": /\\\\\\.js\\$/,
        "use": Array [
          Object {
            "loader": "babel-loader",
            "options": Object {
              "babelrc": false,
              "plugins": Array [
                "transform-flow-strip-types",
              ],
              "presets": Array [
                Array [
                  "es2015",
                  Object {
                    "modules": false,
                  },
                ],
                "react",
                "stage-0",
              ],
            },
          },
        ],
      },
      Object {
        "test": /\\\\\\.\\(s\\?css\\)\\$/,
        "use": Array [
          Object {
            "loader": "/project/packages/gluestick/node_modules/extract-text-webpack-plugin/loader.js",
            "options": Object {
              "omit": 1,
              "remove": true,
            },
          },
          Object {
            "loader": "style-loader",
          },
          Object {
            "loader": "css-loader",
            "options": Object {
              "minimize": false,
              "sourceMap": false,
            },
          },
          Object {
            "loader": "postcss-loader",
          },
          Object {
            "loader": "sass-loader",
            "options": Object {
              "outputStyle": "expanded",
              "sourceMap": false,
              "sourceMapContents": true,
            },
          },
        ],
      },
      Object {
        "test": /\\\\\\.\\(png\\|jpg\\|gif\\|ico\\|svg\\)\\(\\\\\\?v=\\\\d\\+\\\\\\.\\\\d\\+\\\\\\.\\\\d\\+\\)\\?\\$/,
        "use": Array [
          Object {
            "loader": "/project/packages/gluestick/node_modules/extract-text-webpack-plugin/loader.js",
            "options": Object {
              "omit": 1,
              "remove": true,
            },
          },
          Object {
            "loader": "file-loader",
            "options": Object {
              "name": "[name]-[hash].[ext]",
            },
          },
          Object {
            "loader": "image-webpack-loader",
            "options": Object {},
          },
        ],
      },
      Object {
        "test": /\\\\\\.\\(woff\\|woff2\\|ttf\\|eot\\)\\(\\\\\\?v=\\\\d\\+\\\\\\.\\\\d\\+\\\\\\.\\\\d\\+\\)\\?\\$/,
        "use": Array [
          Object {
            "loader": "file-loader",
            "options": Object {
              "name": "[name]-[hash].[ext]",
            },
          },
        ],
      },
    ],
  },
  "node": Object {
    "net": "empty",
  },
  "output": Object {
    "chunkFilename": "[name].[hash].js",
    "filename": "[name].[hash].js",
    "path": "/project/build/assets",
    "publicPath": "/assets/",
  },
  "plugins": Array [
    NormalModuleReplacementPlugin {
      "newResource": "/project/packages/gluestick/src/config/webpack/mocks/nullMock.js",
      "resourceRegExp": /\\\\\\.server\\(\\\\\\.js\\)\\?\\$/,
    },
    DuplicatePackageCheckerPlugin {
      "options": Object {
        "emitError": false,
        "exclude": null,
        "verbose": false,
      },
    },
    ProgressPlugin {
      "handler": undefined,
      "profile": undefined,
    },
    CommonsChunkPlugin {
      "async": undefined,
      "children": undefined,
      "chunkNames": Array [
        "vendor",
      ],
      "deepChildren": undefined,
      "filenameTemplate": "vendor-[hash].bundle.js",
      "ident": "/project/packages/gluestick/node_modules/webpack/lib/optimize/CommonsChunkPlugin.js2",
      "minChunks": undefined,
      "minSize": undefined,
      "selectedChunks": undefined,
    },
    ChunksPlugin {
      "configuration": Object {
        "context": "/project",
        "entry": Object {},
        "module": Object {
          "rules": Array [
            Object {
              "exclude": /\\(node_modules\\\\/\\(\\?!gluestick\\)\\)\\|\\(build\\\\/assets\\)/,
              "test": /\\\\\\.js\\$/,
              "use": Array [
                Object {
                  "loader": "babel-loader",
                  "options": Object {
                    "babelrc": false,
                    "plugins": Array [
                      "transform-flow-strip-types",
                    ],
                    "presets": Array [
                      Array [
                        "es2015",
                        Object {
                          "modules": false,
                        },
                      ],
                      "react",
                      "stage-0",
                    ],
                  },
                },
              ],
            },
            Object {
              "test": /\\\\\\.\\(s\\?css\\)\\$/,
              "use": Array [
                "style-loader",
                Object {
                  "loader": "css-loader",
                  "options": Object {
                    "minimize": false,
                    "sourceMap": false,
                  },
                },
                Object {
                  "loader": "postcss-loader",
                },
                Object {
                  "loader": "sass-loader",
                  "options": Object {
                    "outputStyle": "expanded",
                    "sourceMap": false,
                    "sourceMapContents": true,
                  },
                },
              ],
            },
            Object {
              "test": /\\\\\\.\\(png\\|jpg\\|gif\\|ico\\|svg\\)\\(\\\\\\?v=\\\\d\\+\\\\\\.\\\\d\\+\\\\\\.\\\\d\\+\\)\\?\\$/,
              "use": Array [
                Object {
                  "loader": "file-loader",
                  "options": Object {
                    "name": "[name]-[hash].[ext]",
                  },
                },
                Object {
                  "loader": "image-webpack-loader",
                  "options": Object {},
                },
              ],
            },
            Object {
              "test": /\\\\\\.\\(woff\\|woff2\\|ttf\\|eot\\)\\(\\\\\\?v=\\\\d\\+\\\\\\.\\\\d\\+\\\\\\.\\\\d\\+\\)\\?\\$/,
              "use": Array [
                Object {
                  "loader": "file-loader",
                  "options": Object {
                    "name": "[name]-[hash].[ext]",
                  },
                },
              ],
            },
          ],
        },
        "node": Object {
          "net": "empty",
        },
        "output": Object {
          "chunkFilename": "[name].[hash].js",
          "filename": "[name].[hash].js",
          "path": "/project/build/assets",
          "publicPath": "/assets/",
        },
        "plugins": Array [
          NormalModuleReplacementPlugin {
            "newResource": "/project/packages/gluestick/src/config/webpack/mocks/nullMock.js",
            "resourceRegExp": /\\\\\\.server\\(\\\\\\.js\\)\\?\\$/,
          },
          DuplicatePackageCheckerPlugin {
            "options": Object {
              "emitError": false,
              "exclude": null,
              "verbose": false,
            },
          },
          ProgressPlugin {
            "handler": undefined,
            "profile": undefined,
          },
          CommonsChunkPlugin {
            "async": undefined,
            "children": undefined,
            "chunkNames": Array [
              "vendor",
            ],
            "deepChildren": undefined,
            "filenameTemplate": "vendor-[hash].bundle.js",
            "ident": "/project/packages/gluestick/node_modules/webpack/lib/optimize/CommonsChunkPlugin.js2",
            "minChunks": undefined,
            "minSize": undefined,
            "selectedChunks": undefined,
          },
        ],
        "resolve": Object {
          "alias": Object {
            "apps": "/project/src/apps",
            "assets": "/project/assets",
            "compiled": "/project/node_modules",
            "config": "/project/src/config",
            "root": "/project",
            "shared": "/project/src/shared",
            "src": "/project/src",
          },
          "extensions": Array [
            ".js",
            ".css",
            ".json",
          ],
        },
      },
      "options": Object {},
    },
    ExtractTextPlugin {
      "filename": "[name]-[chunkhash].css",
      "id": 1,
      "options": Object {},
    },
    OptimizeCssAssetsPlugin {
      "options": Object {
        "assetNameRegExp": /\\\\\\.css\\$/g,
        "canPrint": false,
        "cssProcessor": [Function],
        "cssProcessorOptions": Object {},
      },
    },
    DefinePlugin {
      "definitions": Object {
        "process.env.NODE_ENV": "\\"production\\"",
      },
    },
    UglifyJsPlugin {
      "options": Object {
        "compress": Object {
          "warnings": false,
        },
        "sourceMap": true,
      },
    },
    NormalModuleReplacementPlugin {
      "newResource": "/project/packages/gluestick/src/config/webpack/mocks/emptyObjMock.js",
      "resourceRegExp": /gluestick\\\\/shared\\\\/lib\\\\/errorUtils/,
    },
  ],
  "resolve": Object {
    "alias": Object {
      "apps": "/project/src/apps",
      "assets": "/project/assets",
      "compiled": "/project/node_modules",
      "config": "/project/src/config",
      "root": "/project",
      "shared": "/project/src/shared",
      "src": "/project/src",
    },
    "extensions": Array [
      ".js",
      ".css",
      ".json",
    ],
  },
}
`;

exports[`config/compileWebpackConfig in production environment returns the correct server config 1`] = `
Object {
  "bail": true,
  "context": "/project",
  "devtool": "source-map",
  "entry": Object {
    "renderer": "/project/packages/gluestick/src/renderer/entry.js",
  },
  "externals": Array [
    [Function],
  ],
  "module": Object {
    "noParse": Array [
      /cli\\\\/helpers/,
    ],
    "rules": Array [
      Object {
        "exclude": /\\(node_modules\\\\/\\(\\?!gluestick\\)\\)\\|\\(build\\\\/assets\\)/,
        "test": /\\\\\\.js\\$/,
        "use": Array [
          Object {
            "loader": "babel-loader",
            "options": Object {
              "babelrc": false,
              "plugins": Array [
                "transform-flow-strip-types",
              ],
              "presets": Array [
                "es2015",
                "react",
                "stage-0",
              ],
            },
          },
        ],
      },
      Object {
        "test": /\\\\\\.\\(s\\?css\\)\\$/,
        "use": Array [
          "ignore-loader",
        ],
      },
      Object {
        "test": /\\\\\\.\\(png\\|jpg\\|gif\\|ico\\|svg\\)\\(\\\\\\?v=\\\\d\\+\\\\\\.\\\\d\\+\\\\\\.\\\\d\\+\\)\\?\\$/,
        "use": Array [
          Object {
            "loader": "file-loader",
            "options": Object {
              "emitFile": false,
              "name": "[name]-[hash].[ext]",
            },
          },
          Object {
            "loader": "image-webpack-loader",
            "options": Object {},
          },
        ],
      },
      Object {
        "test": /\\\\\\.\\(woff\\|woff2\\|ttf\\|eot\\)\\(\\\\\\?v=\\\\d\\+\\\\\\.\\\\d\\+\\\\\\.\\\\d\\+\\)\\?\\$/,
        "use": Array [
          Object {
            "loader": "file-loader",
            "options": Object {
              "emitFile": false,
              "name": "[name]-[hash].[ext]",
            },
          },
        ],
      },
    ],
  },
  "node": Object {
    "__dirname": false,
    "__filename": false,
    "net": "empty",
  },
  "output": Object {
    "chunkFilename": "[name].js",
    "filename": "[name].js",
    "libraryTarget": "commonjs2",
    "path": "/project/build/server",
    "pathinfo": true,
  },
  "plugins": Array [
    ProgressPlugin {
      "handler": undefined,
      "profile": undefined,
    },
    DefinePlugin {
      "definitions": Object {
        "process.env.NODE_ENV": "\\"production\\"",
      },
    },
    LimitChunkCountPlugin {
      "options": Object {
        "maxChunks": 1,
      },
    },
  ],
  "resolve": Object {
    "alias": Object {
      "application-config": "/project/src/config/application",
      "apps": "/project/src/apps",
      "assets": "/project/assets",
      "compiled": "/project/node_modules",
      "config": "/project/src/config",
      "entry-wrapper": "/project/gluestick/EntryWrapper",
      "gluestick-hooks": "/project/src/gluestick.hooks.js",
      "plugins-config-path": undefined,
      "project-entries": "/project/gluestick/entries",
      "project-entries-config": "/project/src/entries.json",
      "react": "/project/packages/gluestick/node_modules/react/cjs/react.production.min.js",
      "react-dom": "/project/packages/gluestick/node_modules/react-dom/cjs/react-dom.production.min.js",
      "react-dom/server": "/project/packages/gluestick/node_modules/react-dom/cjs/react-dom-server.node.production.min.js",
      "redux-middlewares": "/project/src/config/redux-middleware",
      "root": "/project",
      "shared": "/project/src/shared",
      "src": "/project/src",
    },
    "extensions": Array [
      ".js",
      ".css",
      ".json",
    ],
  },
  "target": "node",
}
`;
